lib/rugged_adapter/git_layer_rugged.rb

```diff
@@ -188,13 +188,29 @@ def cat_file(options, sha)
         @repo.lookup(sha).read_raw
       end
 
-      def apply_patch(head_sha = 'HEAD', patch=nil)
-        false # Rewrite gollum-lib's revert so that it doesn't require a direct equivalent of Grit's apply_patch
-      end
-
-      def revert(path, sha1, sha2, ref)
-        # FIXME: See https://github.com/gollum/grit_adapter/pull/14
-        fail NotImplementedError
+      def revert(path, sha1, sha2)
+        options = {}
+        options[:paths] = [path] if path
```

Comment:
What happens when `options[:paths] = [nil]`? Would this throw an error? Otherwise you could simplify the initialization of the options hash to `options = {paths: [path], location: :index}`.

Reply:
Good question. 👍 Will investigate.